File: Easy/557. Reverse Words in a String III/557. Reverse Words in a String III.java
class Solution {
    public String reverseWords(String s) {
        // Split the input string into words using whitespace as a delimiter
        String[] words = s.split(" ");

        // Create a StringBuilder to store the reversed words
        StringBuilder reversedWords = new StringBuilder();

        // Iterate through the words and reverse each word
        for (String word : words) {
            // Reverse the characters of the word and append it to reversedWords
            String reversedWord = new StringBuilder(word).reverse().toString();
            reversedWords.append(reversedWord).append(" ");
        }

        // Remove the trailing whitespace and convert StringBuilder to String
        String result = reversedWords.toString().trim();

        return result;
    }
}

File: Easy/557. Reverse Words in a String III/557. Reverse Words in a String III.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    // Split the input string into words using whitespace as a delimiter
    const words = s.split(' ');

    // Create an array to store the reversed words
    const reversedWords = [];

    // Iterate through the words and reverse each word
    for (const word of words) {
        // Reverse the characters of the word and push it to the reversedWords array
        const reversedWord = word.split('').reverse().join('');
        reversedWords.push(reversedWord);
    }

    // Join the reversed words with whitespace to form the final sentence
    const result = reversedWords.join(' ');

    return result;
};
